### PR TITLE
getReader().read() only returns the first line

### LINKED ISSUE #270: `transform:`
@mxcl: the `transform:` key seems like it might be behaving oddly. from the [build logs](https://github.com/pkgxdev/pantry/actions/runs/7506699325/job/20438689379?pr=4830) of https://github.com/pkgxdev/pantry/pull/4830:

```
{ v: "jdk-17.0.1+7", rv: "17.0.1.7" }
{ v: "jdk-17.0.1+8", rv: "17.0.1.8" }
{ v: "jdk-17.0.1+9", rv: "17.0.1.9" }
{ v: "jdk-17.0.1+10", rv: "17.0.1.10" }
{ v: "jdk-17-ga", rv: "17-ga" }
{ v: "jdk-17.0.1+11", rv: "17.0.1.11" }
{ v: "jdk-17.0.1+12", rv: "17.0.1.12" }
{ v: "jdk-17.0.2+1", rv: "17.0.2.1" }
{ v: "jdk-17.0.2+2", rv: "17.0.2.2" }
{ v: "jdk-17.0.1-ga", rv: "17.0.1-ga" }
{ v: "jdk-17.0.2+3", rv: "17.0.2.3" }
{ v: "jdk-17.0.2+4", rv: "17.0.2.4" }
{ v: "jdk-17.0.2+5", rv: "17.0.2.5" }
{ v: "jdk-17.0.2+6", rv: "17.0.2.6" }
{ v: "jdk-17.0.2+7", rv: "17.0.2.7" }
```
but it seems to be selecting the _first_ one `17.0.1.7` for `^17`, rather than `17.0.10.6`.

as an aside, should `strip:` and `transform:` be mutually exclusive? it looks like they're handled `transform` > `strip`, which surprised me, but there's no real reason it should have. as long as `version.tag` is still preserved, all should be fine. 🤞

### PR DESCRIPTION
**Pull request by jhheider**
closes #270

### CODE REVIEW

**Comment by mxcl** [comment]
lol whoops. I swear I printed out the output from the handler and there was lots. So maybe it isn't always the first line?

File: lib/hooks/usePantry.getVersions.ts [hunk 1]
```diff
@@ -305,9 +305,7 @@ async function handleTransformer(transform: string, versions: string[]): Promise
     `));
   await writer.close()
 
-  const read = cmd.stdout.getReader().read().then(x => x.value)
-
-  const [out, { success: ok }] = await Promise.all([read, cmd.status])
+  const { stdout: out, success: ok } = await cmd.output()
```
Comment (mxcl) [inline lib/hooks/usePantry.getVersions.ts:308]: I think the docs imply you should await cmd.status but if it works maybe not then.

Comment (jhheider) [reply lib/hooks/usePantry.getVersions.ts:308]: Pulled that example out of the current docs, I believe. I'll add the reference.

Comment (jhheider) [reply lib/hooks/usePantry.getVersions.ts:308]: https://docs.deno.com/runtime/tutorials/subprocess#simple-example

**Comment by jhheider** [comment]
> lol whoops. I swear I printed out the output from the handler and there was lots. So maybe it isn't always the first line?

i think sometimes i got 3, so it might be a race condition of some flavor.

**Pull request merged**